packages/cosmic-swingset/tools/test-kit.js

```diff
@@ -214,49 +217,54 @@ export const makeCosmicSwingsetTestKit = async (
   if (initMessage.isBootstrap === undefined && !swingStore) {
     initMessage.isBootstrap = true;
   }
+  const portNums = extractPortNums(initMessage);
+  const knownPorts = new Map(
+    Object.entries(portNums).map(([name, value]) => [value, name]),
+  );
 
   if (!swingStore) swingStore = initSwingStore(); // in-memory
   const { hostStorage } = swingStore;
 
-  const actionQueueStorage = makeQueueStorageMock().storage;
-  const highPriorityQueueStorage = makeQueueStorageMock().storage;
-  const { kvStore: mailboxKVStore, ...mailboxBufferMethods } =
-    makeBufferedStorage(
-      /** @type {KVStore<Mailbox>} */ (makeKVStoreFromMap(new Map())),
-    );
-  const mailboxStorage = { ...mailboxKVStore, ...mailboxBufferMethods };
-
-  const savedChainSends = [];
-  const clearChainSends = async () => savedChainSends.splice(0);
-  const replayChainSends = (..._args) => {
-    throw Error('not implemented');
+  const fakeAgcc = {
+    /** @type {(destPort: unknown, msgJson: string) => string} */
+    send: (destPort, msgJson) => {
+      const portName =
+        knownPorts.get(destPort) || Fail`unknown cosmos port ${destPort}`;
+      const msg = JSON.parse(msgJson);
+      const result = receiveBridgeSend(portName, msg);
+      return JSON.stringify(result);
+    },
   };
+  const actionQueueStorage = makeQueueStorage(
+    msg => fakeAgcc.send(portNums.storage, msg),
+    STORAGE_PATH.ACTION_QUEUE,
+  );
+  const highPriorityQueueStorage = makeQueueStorage(
+    msg => fakeAgcc.send(portNums.storage, msg),
+    STORAGE_PATH.HIGH_PRIORITY_QUEUE,
+  );
 
-  if (!slogSender && (env.SLOGFILE || env.SLOGSENDER)) {
-    slogSender = await makeSlogSender({ env });
-  }
-
-  const launchResult = await launch({
-    swingStore,
-    actionQueueStorage,
-    highPriorityQueueStorage,
-    mailboxStorage,
-    clearChainSends,
-    replayChainSends,
-    bridgeOutbound: receiveBridgeSend,
-    vatconfig: config,
-    argv: { bootMsg: makeInitMsg(initMessage) },
+  const { name: dbDir, removeCallback: cleanupDB } = tmp.dirSync({
+    prefix: debugName || 'testdb',
+    unsafeCleanup: true,
+  });
+  const launchChain = makeLaunchChain(fakeAgcc, dbDir, {
```

Comment:
I'm pretty sure that `stateDBDir` is unused if `slogSender` and `swingStore` are provided.